Funcky/Internal/PartitionBuilder.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections.Immutable;
+
+namespace Funcky.Internal;
+
+internal sealed class PartitionBuilder<TLeft, TRight>
+{
+    private ImmutableArray<TLeft>.Builder? _left;
```

Comment:
How can this be null at any point, it is created in the constructor?

Reply:
Oh, I forgot to clean this up. Fixed here: 41fa39a19c27ebd41eb7ad04f5e34eb1f267fda5